### PR TITLE
feat(mcp): expose server instructions from InitializeResult on MCPClient

### LINKED ISSUE #1813: [FEATURE]  Expose MCP Server `instructions` from `InitializeResult` on `MCPClient`
### Problem Statement

`MCPClient._start_session()` calls `session.initialize()` but discards the `InitializeResult`. This means Host applications have no
 way to access the server's `instructions` field — a standard MCP feature for providing LLMs with contextual knowledge about tool interdependencies and operational constraints.

```python
# mcp_client.py line ~504
await session.initialize()  # InitializeResult (including instructions) is discarded
```

MCP clients like Claude Code, VSCode, and Goose inject server instructions into the LLM's system prompt. Strands-based Host applications cannot do this because the data is inaccessible.

### Proposed Solution
Store `InitializeResult.instructions` on the `MCPClient` instance:

```python
# In MCPClient._start_session():
result = await session.initialize()
self.server_instructions = result.instructions or ""
```

Host applications can then access it:

```python
with mcp_client:
    instructions = mcp_client.server_instructions
    agent = Agent(
        system_prompt=f"...\n{instructions}",
        tools=mcp_client.list_tools_sync(),
    )
```

### Use Case

I'm building an Agent (Host) that connects to an MCP Server with detailed workflow instructions (multi-phase slide generation). The server returns these via the standard `instructions` field in `InitializeResult`, but my Agent cannot access them through `MCPClient`.

Without this, I have to either:
- Hardcode the MCP Server's instructions in the Agent's system prompt (breaks DRY, drifts)
- Make a separate HTTP call to the MCP Server just to get instructions (wasteful)

Per the MCP spec, the Host is responsible for deciding how to use instructions — but the Client (MCPClient) needs to make them accessible first.

References:
- MCP spec: https://modelcontextprotocol.io/specification/2025-06-18/basic/lifecycle
- MCP blog on server instructions: https://blog.modelcontextprotocol.io/posts/2025-11-03-using-server-instructions/

### Alternatives Solutions

- Host could make a separate `initialize` HTTP request to get instructions, but this creates a redundant connection
- Instructions could be duplicated in every tool description, but this wastes context window and violates MCP best practices

### Additional Context

The fix is ~3 lines: store the result of `session.initialize()` and expose `instructions` as a public attribute. The SDK's responsibility is only to　make the data accessible — how/whether to inject into the system prompt remains the Host's decision.

### PR DESCRIPTION
**Pull request by ShotaroKataoka**
## Description

Store `InitializeResult.instructions` on `MCPClient.server_instructions` so Host applications can access MCP server instructions and inject them into the LLM's system prompt.

The [MCP specification](https://modelcontextprotocol.io/specification/2025-06-18/basic/lifecycle) defines `instructions` in `InitializeResult` for servers to provide LLMs with contextual knowledge about tool interdependencies and operational constraints. Currently `MCPClient` discards the `InitializeResult`, making instructions inaccessible to Host applications.

Changes:
- `mcp_client.py`: Store `init_result.instructions` after `session.initialize()` (3 lines added)
- `conftest.py`: Fix mock to return proper `InitializeResult` with `instructions` field
- `test_mcp_client.py`: Add 2 tests (default empty string, populated from server)

The SDK's responsibility is only to make the data accessible — how/whether to inject into the system prompt remains the Host's decision.

## Related Issues

Closes #1813

## Documentation PR

N/A — Public attribute `server_instructions` is self-documenting. Docs update can follow if needed.

## Type of Change

New feature

## Testing

- 2106 passed, 2 failed (pre-existing failures unrelated to this PR — region-dependent default model prefix `apac` vs `us`)
- 128 MCP tests all pass
- 2 new tests added

## Checklist
- [x] I have read the CONTRIBUTING document
- [x] I have added any necessary tests that prove my fix is effective or my feature works
- [ ] I have updated the documentation accordingly
- [x] I have added an appropriate example to the documentation to outline the feature, or no new docs are needed
- [x] My changes generate no new warnings
- [x] Any dependent changes have been merged and published

### CODE REVIEW

File: src/strands/tools/mcp/mcp_client.py [hunk 1]
```diff
@@ -732,9 +733,11 @@ async def _async_background_thread(self) -> None:
                     elicitation_callback=self._elicitation_callback,
                 ) as session:
                     self._log_debug_with_thread("initializing MCP session")
-                    await session.initialize()
+                    init_result = await session.initialize()
 
                     self._log_debug_with_thread("session initialized successfully")
+                    # Store server instructions from InitializeResult for Host applications
+                    self.server_instructions = init_result.instructions or ""
```
Comment (dbschmigelski) [inline src/strands/tools/mcp/mcp_client.py]: PR looks great to me. just requesting couple nit changes

1. Lets make server_instructions str | None so we can differentiate between instructions not present and the server actually declaring it as empty string
2. lets move the public attribute  server_instructions above the private ones like self._consumers: set[Any] = set()

Comment (ShotaroKataoka) [reply src/strands/tools/mcp/mcp_client.py]: > PR looks great to me. just requesting couple nit changes
> 
> 1. Lets make server_instructions str | None so we can differentiate between instructions not present and the server actually declaring it as empty string
> 2. lets move the public attribute  server_instructions above the private ones like self._consumers: set[Any] = set()

Thanks for the review! Addressed both points:

1. `server_instructions: str | None = None` — `None` means server didn't provide instructions, `""` means explicitly empty
2. Moved `server_instructions` above private attributes (`_consumers`, etc.)

Updated test accordingly (`assert client.server_instructions is None`).

Force-pushed: ba91fba

**Review by mehtarac — APPROVED** [review]
